store/index/index.go

```diff
@@ -18,16 +21,48 @@ import (
 The format of that append only log is:
 
 ```text
-    |                  Once              |                    Repeated                 |
-    |                                    |                                             |
-    |       4 bytes      | Variable size |         4 bytes        |  Variable size | … |
-    | Size of the header |   [`Header`]  | Size of the Recordlist |   Recordlist   | … |
+    |                    Repeated                 |
+    |                                             |
+    |         4 bytes        |  Variable size | … |
+    | Size of the Recordlist |   Recordlist   | … |
 ```
 */
-const IndexVersion uint8 = 2
+
+// In-memory buckets are used to track the location of records within the index
+// files. The buckets map a bit-prefix to a bucketPos value.  The bucketPos
+// encodes both the index file number and the record offset within that
+// file. If 1GiB is the maximum size for a file, then the local data offset is
+// kept in the first GiB worth of bits (30) of the bucketPos.  The file number
+// is kept in the bits above that.  It is necessary for the file number to wrap
+// before it reaches a value greater than the number of bits available to
+// record it in the buckerPos.  This results in a trade-off between allowing
+// larger files or allowing more files, but with the same overall maximum
+// storage.
+//
+// With a 1GiB local offset taking the first 30 bits of a 64 bit number, that
+// leaves 34 bits left to encode the file number.  Instead of having logic to
+// wrap the file number at the largest value allowed by the available bits, the
+// file number is represented as a 32-bit value that always wraps at 4GiB.
```

Comment:
I find it confusing to mix the number of files with a number of bytes unit. Perhaps writing something like "wraps at 2^32 files" makes it clearer? It might be just me, so feel free to leave it like that. (There's another occurrence further down this comment).